test/split_tests.cc

```diff
@@ -0,0 +1,306 @@
+	for(size_t j = 0; j < chunk_ranges.size(); ++j) {
+		const auto& [height, widths] = chunk_ranges[j];
+		for(size_t i = 0; i < widths.size(); ++i) {
+			const auto& chnk = split_chunks[j * chunk_ranges[0].second.size() + i];
+			REQUIRE_LOOP(chnk.offset == offset);
+			REQUIRE_LOOP(chnk.range[0] == height);
+			REQUIRE_LOOP(chnk.range[1] == widths[i]);
+			REQUIRE_LOOP(chnk.global_size == full_chunk.global_size);
+			offset[1] += widths[i];
+		}
+		offset[1] = full_chunk.offset[1];
+		offset[0] += height;
+	}
+}
+
+} // namespace
+
+TEST_CASE("split_1d creates evenly sized chunks if possible", "[split]") {
+	const auto full_chunk = make_full_chunk<1>({128});
+	const auto chunks = split_1d(full_chunk, ones, 4);
+	check_1d_split(full_chunk, chunks, {32, 32, 32, 32});
+}
+
+TEST_CASE("split_1d distributes remainder evenly", "[split]") {
+	const auto full_chunk = make_full_chunk<1>({13});
+	const auto chunks = split_1d(full_chunk, ones, 5);
+	check_1d_split(full_chunk, chunks, {3, 3, 3, 2, 2});
+}
+
+TEST_CASE("split_1d respects granularity constraints", "[split]") {
+	const auto full_chunk = make_full_chunk<1>({80});
+	const auto chunks = split_1d(full_chunk, {16, 1, 1}, 3);
+	check_1d_split(full_chunk, chunks, {32, 32, 16});
+}
+
+TEST_CASE("split_1d creates fewer chunks than requested if mandated by granularity", "[split]") {
+	const auto full_chunk = make_full_chunk<1>({96});
+	const auto chunks = split_1d(full_chunk, {48, 1, 1}, 3);
+	check_1d_split(full_chunk, chunks, {48, 48});
+}
+
+TEST_CASE("split_1d preserves offset of original chunk", "[split]") {
+	const auto full_chunk = chunk<3>{{37, 42, 7}, {128, 1, 1}, {128, 1, 1}};
+	const auto chunks = split_1d(full_chunk, ones, 4);
+
+	CHECK(chunks[0].offset == id<3>{37 + 0, 42, 7});
+	CHECK(chunks[1].offset == id<3>{37 + 32, 42, 7});
+	CHECK(chunks[2].offset == id<3>{37 + 64, 42, 7});
+	CHECK(chunks[3].offset == id<3>{37 + 96, 42, 7});
+
+	check_1d_split(full_chunk, chunks, {32, 32, 32, 32});
+}
+
+TEST_CASE("split_1d preserves ranges of original chunk in other dimensions", "[split]") {
+	const auto full_chunk = make_full_chunk<3>({128, 42, 341});
+	const auto chunks = split_1d(full_chunk, ones, 4);
+	for(size_t i = 0; i < 4; ++i) {
+		REQUIRE_LOOP(chunks[0].range == range<3>{32, 42, 341});
+	}
+}
+
+TEST_CASE("split_2d produces perfectly square chunks if possible", "[split]") {
+	const auto full_chunk = make_full_chunk<2>({128, 128});
+	const auto chunks = split_2d(full_chunk, ones, 4);
+	check_2d_split(full_chunk, chunks,
+	    {
+	        {64, {64, 64}}, //
+	        {64, {64, 64}}  //
+	    });
+}
+
+TEST_CASE("split_2d respects granularity constraints") {
+	SECTION("simple constrained split") {
+		const auto full_chunk = make_full_chunk<2>({128, 128});
+		const auto chunks = split_2d(full_chunk, {8, 8, 1}, 8);
+		check_2d_split(full_chunk, chunks,
+		    {
+		        {32, {64, 64}}, //
+		        {32, {64, 64}}, //
+		        {32, {64, 64}}, //
+		        {32, {64, 64}}  //
```

Comment:
In the other tests, I think the `,` at the last line forces clang-format to format it as intended, but here you use `//`. If this is accurate, then for consistency I'd change this and the previous instance to also use the `,` style.

Reply:
Ah, that's why, I wrote those tests a while ago and was confused why seemingly only some needed the comments. Will fix.